pyodide-build/pyodide_build/buildall.py

```diff
@@ -264,10 +290,43 @@ def build_from_graph(pkg_map: Dict[str, BasePackage], outputdir: Path, args) ->
     # dependents, because the ordering ought not to change after insertion.
     build_queue: PriorityQueue = PriorityQueue()
 
-    print("Building the following packages: " + ", ".join(sorted(pkg_map.keys())))
+    if args.force:
+        # If "force" is set, just rebuild everything
+        needs_build = set(pkg_map.keys())
+        already_built = set()
+    else:
+        needs_build = set()
+
+        for pkg in pkg_map.values():
+            # Otherwise, rebuild packages that have been updated and their dependents.
+            if needs_rebuild(pkg.pkgdir, pkg.pkgdir / "build", pkg.meta):
+                mark_package_needs_build(pkg_map, pkg, needs_build)
+
+        # We won't rebuild the complement of the packages that we will build.
+        already_built = set(pkg_map.keys()).difference(needs_build)
+        for pkg_name in already_built:
+            # Hacky way to prevent these from being rebuilt: add a dependency
+            # that won't ever be satisfied
+            pkg_map[pkg_name].unbuilt_dependencies.add("!don't build this")
```

Comment:
That's a bit ugly, can't we rather remove it from needs_build ?

Reply:
Well they already aren't in `needs_build` since we just calculated `already_built` as the complement of `needs_build`. Let me see what happens if I remove this.

Reply:
Yeah it works fine without this, which makes sense given the logic below.